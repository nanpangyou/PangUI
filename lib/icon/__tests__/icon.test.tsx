import * as React from "react";
import * as renderer from "react-test-renderer";
import { mount } from "enzyme";
import Icon from "../icon";
describe("icon组件", () => {
  it("icon存在", () => {
    expect(Icon).toBeTruthy;
  });
  it("icon渲染 successfully", () => {
    const json = renderer.create(<Icon name="loading"></Icon>).toJSON();
    expect(json).toMatchSnapshot();
  });
  it("icon的onclick事件", () => {
    const fn = jest.fn();
    const component = mount(<Icon name="setting" onClick={fn}></Icon>);
    component.find("svg").simulate("click");
    expect(fn).toBeCalled();
  });
});
