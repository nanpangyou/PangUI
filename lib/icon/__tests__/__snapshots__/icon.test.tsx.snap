// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon组件 icon渲染 1`] = `
<svg
  className="pang-icon"
>
  <use
    xlinkHref="#loading"
  />
</svg>
`;

exports[`icon组件 icon渲染 successfully 1`] = `
<svg
  className="pang-icon"
>
  <use
    xlinkHref="#loading"
  />
</svg>
`;
